src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -83,14 +101,126 @@ public async Task<IActionResult> Index()
+            {
+                case TenantsOrder.Name:
+                    entries = entries.OrderBy(t => t.Name).ToList();
+                    break;
+                case TenantsOrder.State:
+                    entries = entries.OrderBy(t => t.ShellSettings?.State).ToList();
+                    break;
+                default:
+                    entries = entries.OrderByDescending(t => t.Name).ToList();
+                    break;
+            }
+            var count = entries.Count();
+
+            var results = entries
+                .Skip(pager.GetStartIndex())
+                .Take(pager.PageSize).ToList();
+
+            // Maintain previous route data when generating page links
+            var routeData = new RouteData();
+            routeData.Values.Add("Options.Filter", options.Filter);
+            routeData.Values.Add("Options.Search", options.Search);
+            routeData.Values.Add("Options.SqlProvider", options.SqlProvider);
+            routeData.Values.Add("Options.OrderBy", options.OrderBy);
+
+            var pagerShape = (await New.Pager(pager)).TotalItemCount(count).RouteData(routeData);
+
+            var model = new AdminIndexViewModel
+            {
+                ShellSettingsEntries = results,
+                Options = options,
+                Pager = pagerShape
             };
 
             return View(model);
         }
 
+        [HttpPost]
+        public async Task<IActionResult> Index(BulkActionViewModel model)
+        {
+            foreach(var tenantName in model.TenantNames)
+            {
+                var shellContext = (await GetShellsAsync())
+                    .Where(x => string.Equals(x.Settings.Name, tenantName, StringComparison.OrdinalIgnoreCase))
+                    .FirstOrDefault();
+
+                if (shellContext == null)
+                {
+                    break;
+                }
+
+                var shellSettings = shellContext.Settings;
+
+                switch (model.BulkAction.ToString())
+                {
+                    case "Disable":
+                        if (string.Equals(shellSettings.Name, ShellHelper.DefaultShellName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            _notifier.Error(H["You cannot disable the default tenant."]);
+                        }
+
+                        if (shellSettings.State != TenantState.Running)
+                        {
+                            _notifier.Error(H["You can only disable an Enabled tenant."]);
+                        }
+
+                        shellSettings.State = TenantState.Disabled;
+                        await _orchardHost.UpdateShellSettingsAsync(shellSettings);
+
+                        break;
+
+                    case "Enable":
+                        if (shellSettings.State != TenantState.Disabled)
+                        {
+                            _notifier.Error(H["You can only enable a Disabled tenant."]);
```

Comment:
Add the tenant name and use Warning. Nothing broke, just nothing was done. The name is useful because it's a batch, so the message could appear for many tenants, or even if only one we would have no idea which one.

Reply:
OK, good remark. Changes done.